jdtls.ext/com.microsoft.java.maven.plugin/src/main/java/com/microsoft/java/maven/AddDependencyHandler.java

```diff
@@ -0,0 +1,266 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.microsoft.java.maven;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite;
+import org.eclipse.jdt.core.manipulation.CodeStyleConfiguration;
+import org.eclipse.jdt.core.refactoring.CompilationUnitChange;
+import org.eclipse.jdt.internal.core.manipulation.dom.ASTResolving;
+import org.eclipse.jdt.ls.core.internal.ChangeUtil;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.xml.sax.Attributes;
+import org.xml.sax.Locator;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AddDependencyHandler {
+
+    public static class AddDependencyParams {
+
+        final String fullClassName;
+        final String artifactInfo;
+        final String uri;
+        final int line;
+        final int character;
+        final int length;
+
+        public AddDependencyParams(String fullClassName, String artifactInfo, 
+                String uri, int line, int character, int length) {
+            this.fullClassName = fullClassName;
+            this.artifactInfo = artifactInfo;
+            this.uri = uri;
+            this.line = line;
+            this.character = character;
+            this.length = length;
+        }
+    }
+
+    private static WorkspaceEdit importEdit(AddDependencyParams params, 
+            ICompilationUnit unit, CompilationUnit astRoot) {
+        // import the new class
+        final CompilationUnitChange cuChange = new CompilationUnitChange("", unit);
+        final ImportRewrite importRewrite = CodeStyleConfiguration.createImportRewrite(astRoot, true);
+        importRewrite.addImport(params.fullClassName);
+        try {
+            cuChange.setEdit(importRewrite.rewriteImports(null));
+            return ChangeUtil.convertToWorkspaceEdit(cuChange);
+        } catch (CoreException e) {
+            return new WorkspaceEdit();
```

Comment:
reason to swallow the exception? suggest to at least do some logging in case of future diagnose

Reply:
to do